virttest/utils_libvirt/libvirt_vmxml.py

```diff
@@ -57,6 +57,26 @@ def check_guest_xml(vm_name, pat_in_dumpxml, option="", status_error=False):
         raise exceptions.TestFail(msg)
 
 
+def check_guest_machine_type(vmxml, expected_version="9.4.0"):
+    """
+    Check guest machine type version
+
+    :param vmxml: the guest xml.
+    :param expected_version: expected guest machine version, eg
+    if <type arch='x86_64' machine='pc-q35-rhel8.2.0'>hvm</type> exists,
+    which means expected_version is 8.2.0
+    :return: True or False, the flag of checking successfully or not.
+    """
+    actual_version = int("".join(re.findall(r"\d+", vmxml.os.machine.split("-")[-1])))
+    expected_version = int(expected_version.replace(".", ""))
+
+    if actual_version < expected_version:
+        LOG.debug("Guest config machine should be >= rhel{}".format(expected_version))
+        return True
```

Comment:
I think we should return False in this condition.

Reply:
Thanks , Updated